File: module_chat/src/main/java/com/yyxnb/module_chat/config/DataConfig.java
package com.yyxnb.module_chat.config;

import com.yyxnb.module_chat.bean.MessageBean;
import com.yyxnb.lib_network.utils.GsonUtils;
import com.yyxnb.lib_utils.FileUtils;
import com.yyxnb.lib_widget.AppUtils;

import java.util.List;

public class DataConfig {

    private volatile static List<MessageBean> messageBeans;

    /**
     * 列表数据
     */
    public static List<MessageBean> getMessageBeans() {
        if (messageBeans == null) {
            String content = FileUtils.parseFile(AppUtils.getApp(), "msg_data.json");
            messageBeans = GsonUtils.jsonToList(content, MessageBean.class);
        }
        return messageBeans;
    }
}

File: module_chat/src/main/java/com/yyxnb/module_chat/config/DataConfig.kt
package com.yyxnb.module_chat.config

import com.yyxnb.module_chat.bean.MessageBean
import com.yyxnb.network.utils.GsonUtils.jsonToList
import com.yyxnb.utils.FileUtils
import com.yyxnb.widget.AppUtils

object DataConfig {
    /**
     * 列表数据
     */
    @JvmStatic
    @Volatile
    var messageBeans: List<MessageBean>? = null
        get() {
            if (field == null) {
                val content = FileUtils.parseFile(AppUtils.app, "msg_data.json")
                field = jsonToList(content, MessageBean::class.java)
            }
            return field
        }
        private set
}